src/crimson/os/seastore/cache.h

```diff
@@ -589,16 +596,38 @@ class Cache {
   /// Invalidate extent and mark affected transactions
   void invalidate(CachedExtent &extent);
 
-  Transaction::get_extent_ret query_cache_for_extent(
-    paddr_t offset,
-    CachedExtentRef *out) {
+  template <typename T>
+  get_extent_ret<T> read_extent(
+    TCachedExtentRef<T>&& extent
```

Comment:
Have it take a pointer and return void.  populate_extent rather than read_extent.